src/gems/simulation/output_values.py

```diff
@@ -242,30 +225,39 @@ def _are_mappings_close(
     lhs_keys = lhs.keys()
     rhs_keys = rhs.keys()
 
-    if (lhs_only_keys := lhs_keys - rhs_keys) and any(
-        not lhs[key].ignore for key in lhs_only_keys
-    ):
-        return False
-
-    elif (rhs_only_keys := rhs_keys - lhs_keys) and any(
-        not rhs[key].ignore for key in rhs_only_keys
-    ):
-        return False
-
-    elif intersect_keys := lhs_keys & rhs_keys:
-        if rel_tol == abs_tol == 0.0:
-            return all(lhs[key] == rhs[key] for key in intersect_keys)
+    # Keys present only on the left
+    lhs_only = lhs_keys - rhs_keys
+    if lhs_only:
+        for key in lhs_only:
+            item = lhs[key]
+            if getattr(item, "ignore", False) is False:
```

Comment:
Possible to get rid of all `getattr` and `hasttr` in the PR with strong typing ?